modules/afsocket/socket-options-inet.c

```diff
@@ -39,6 +40,41 @@
 #define SOL_TCP IPPROTO_TCP
 #endif
 
+#ifdef SO_BINDTODEVICE
+static int
+socket_options_inet_set_interface(SocketOptionsInet *self, gint fd)
+{
+  cap_t saved_caps;
+  int result, rc;
+
+  saved_caps = g_process_cap_save();
+  g_process_enable_cap("cap_net_raw");
+  rc = setsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE, self->interface_name, strlen(self->interface_name));
+  result = (rc < 0) ? errno : 0;
+  g_process_cap_restore(saved_caps);
+  return result;
+}
+
+static gboolean
+socket_options_inet_try_to_set_interface(SocketOptionsInet *self, gint fd)
+{
+  int result = socket_options_inet_set_interface(self, fd);
+  if (result != 0)
+    {
+      msg_error("Can't bind to interface", evt_tag_str("interface", self->interface_name), evt_tag_errno("error", result));
```

Comment:
Seemingly this should signal an error, but for me syslog-ng started successfully, as if I had just omitted the interface option.
https://github.com/balabit/syslog-ng/pull/2389#issuecomment-435584668

Reply:
@furiel Have you tried with root or non-root user?
Because it requires to cap_net_raw, and without superuser, I don't sure, that you can get it.

Reply:
I tried this one with non-root user, so I expected it to fail. The issue here is that syslog-ng starts after the fail detection. When this option is not applied, it can cause connectivity issue, which can go semi-silent (we have an apparent error log, but in similar cases like when binding for an already bound port, syslog-ng does not start at all).